man/man1/ogrinfo.1
.TH "ogrinfo" 1 "Wed Sep 16 2015" "GDAL" \" -*- nroff -*-
.ad l
.nh
.SH NAME
ogrinfo \- ogrinfo 
lists information about an OGR supported data source
.SH "SYNOPSIS"
.PP
.PP
.PP
.nf

ogrinfo [--help-general] [-ro] [-q] [-where restricted_where]
        [-spat xmin ymin xmax ymax] [-geomfield field] [-fid fid]
        [-sql statement] [-dialect dialect] [-al] [-so] [-fields={YES/NO}]
        [-geom={YES/NO/SUMMARY}][--formats]
        datasource_name [layer [layer ...]]

.fi
.PP
.SH "DESCRIPTION"
.PP
The ogrinfo program lists various information about an OGR supported data source to stdout (the terminal)\&.
.PP
.IP "\fB\fB-ro\fP:\fP" 1c
Open the data source in read-only mode\&.  
.IP "\fB\fB-al\fP:\fP" 1c
List all features of all layers (used instead of having to give layer names as arguments)\&. 
.IP "\fB\fB-so\fP:\fP" 1c
Summary Only: suppress listing of features, show only the summary information like projection, schema, feature count and extents\&. 
.IP "\fB\fB-q\fP:\fP" 1c
Quiet verbose reporting of various information, including coordinate system, layer schema, extents, and feature count\&.  
.IP "\fB\fB-where\fP \fIrestricted_where\fP:\fP" 1c
An attribute query in a restricted form of the queries used in the SQL WHERE statement\&. Only features matching the attribute query will be reported\&. 
.IP "\fB\fB-sql\fP \fIstatement\fP:\fP" 1c
Execute the indicated SQL statement and return the result\&. 
.IP "\fB\fB-dialect\fP \fIdialect\fP:\fP" 1c
SQL dialect\&. In some cases can be used to use (unoptimized) OGR SQL instead of the native SQL of an RDBMS by passing OGRSQL\&. Starting with GDAL 1\&.10, the 'SQLITE' dialect can also be used with any datasource\&. 
.IP "\fB\fB-spat\fP \fIxmin ymin xmax ymax\fP:\fP" 1c
The area of interest\&. Only features within the rectangle will be reported\&. 
.IP "\fB\fB-geomfield\fP \fIfield\fP:\fP" 1c
(OGR >= 1\&.11) Name of the geometry field on which the spatial filter operates on\&. 
.IP "\fB\fB-fid\fP \fIfid\fP:\fP" 1c
If provided, only the feature with this feature id will be reported\&. Operates exclusive of the spatial or attribute queries\&. Note: if you want to select several features based on their feature id, you can also use the fact the 'fid' is a special field recognized by OGR SQL\&. So, '-where 'fid in (1,3,5)'' would select features 1, 3 and 5\&. 
.IP "\fB\fB-fields\fP={YES/NO}:\fP" 1c
(starting with GDAL 1\&.6\&.0) If set to NO, the feature dump will not display field values\&. Default value is YES\&. 
.IP "\fB\fB-geom\fP={YES/NO/SUMMARY}:\fP" 1c
(starting with GDAL 1\&.6\&.0) If set to NO, the feature dump will not display the geometry\&. If set to SUMMARY, only a summary of the geometry will be displayed\&. If set to YES, the geometry will be reported in full OGC WKT format\&. Default value is YES\&. 
.IP "\fB\fB--formats\fP:\fP" 1c
List the format drivers that are enabled\&. 
.IP "\fB\fIdatasource_name\fP:\fP" 1c
The data source to open\&. May be a filename, directory or other virtual name\&. See the \fCOGR Vector Formats\fP list for supported datasources\&. 
.IP "\fB\fIlayer\fP:\fP" 1c
One or more layer names may be reported\&. 
.PP
.PP
If no layer names are passed then ogrinfo will report a list of available layers (and their layerwide geometry type)\&. If layer name(s) are given then their extents, coordinate system, feature count, geometry type, schema and all features matching query parameters will be reported to the terminal\&. If no query parameters are provided, all features are reported\&.
.PP
Geometries are reported in OGC WKT format\&.
.SH "EXAMPLE"
.PP
Example reporting all layers in an NTF file: 
.PP
.nf

% ogrinfo wrk/SHETLAND_ISLANDS.NTF
INFO: Open of `wrk/SHETLAND_ISLANDS.NTF'
using driver `UK .NTF' successful.
1: BL2000_LINK (Line String)
2: BL2000_POLY (None)
3: BL2000_COLLECTIONS (None)
4: FEATURE_CLASSES (None)

.fi
.PP
.PP
Example using an attribute query is used to restrict the output of the features in a layer: 
.PP
.nf

% ogrinfo -ro -where 'GLOBAL_LINK_ID=185878' wrk/SHETLAND_ISLANDS.NTF BL2000_LINK
INFO: Open of `wrk/SHETLAND_ISLANDS.NTF'
using driver `UK .NTF' successful.

Layer name: BL2000_LINK
Geometry: Line String
Feature Count: 1
Extent: (419794.100000, 1069031.000000) - (419927.900000, 1069153.500000)
Layer SRS WKT:
PROJCS["OSGB 1936 / British National Grid",
    GEOGCS["OSGB 1936",
        DATUM["OSGB_1936",
            SPHEROID["Airy 1830",6377563.396,299.3249646]],
        PRIMEM["Greenwich",0],
        UNIT["degree",0.0174532925199433]],
    PROJECTION["Transverse_Mercator"],
    PARAMETER["latitude_of_origin",49],
    PARAMETER["central_meridian",-2],
    PARAMETER["scale_factor",0.999601272],
    PARAMETER["false_easting",400000],
    PARAMETER["false_northing",-100000],
    UNIT["metre",1]]
LINE_ID: Integer (6.0)
GEOM_ID: Integer (6.0)
FEAT_CODE: String (4.0)
GLOBAL_LINK_ID: Integer (10.0)
TILE_REF: String (10.0)
OGRFeature(BL2000_LINK):2
  LINE_ID (Integer) = 2
  GEOM_ID (Integer) = 2
  FEAT_CODE (String) = (null)
  GLOBAL_LINK_ID (Integer) = 185878
  TILE_REF (String) = SHETLAND I
  LINESTRING (419832.100 1069046.300,419820.100 1069043.800,419808.300
  1069048.800,419805.100 1069046.000,419805.000 1069040.600,419809.400
  1069037.400,419827.400 1069035.600,419842 1069031,419859.000
  1069032.800,419879.500 1069049.500,419886.700 1069061.400,419890.100
  1069070.500,419890.900 1069081.800,419896.500 1069086.800,419898.400
  1069092.900,419896.700 1069094.800,419892.500 1069094.300,419878.100
  1069085.600,419875.400 1069087.300,419875.100 1069091.100,419872.200
  1069094.600,419890.400 1069106.400,419907.600 1069112.800,419924.600
  1069133.800,419927.900 1069146.300,419927.600 1069152.400,419922.600
  1069153.500,419917.100 1069153.500,419911.500 1069153.000,419908.700
  1069152.500,419903.400 1069150.800,419898.800 1069149.400,419894.800
  1069149.300,419890.700 1069149.400,419890.600 1069149.400,419880.800
  1069149.800,419876.900 1069148.900,419873.100 1069147.500,419870.200
  1069146.400,419862.100 1069143.000,419860 1069142,419854.900
  1069138.600,419850 1069135,419848.800 1069134.100,419843
  1069130,419836.200 1069127.600,419824.600 1069123.800,419820.200
  1069126.900,419815.500 1069126.900,419808.200 1069116.500,419798.700
  1069117.600,419794.100 1069115.100,419796.300 1069109.100,419801.800
  1069106.800,419805.000  1069107.300)

.fi
.PP
.SH "AUTHORS"
.PP
Frank Warmerdam <warmerdam@pobox.com>, Silke Reimer <silke@intevation.de> 


NEWS
= GDAL/OGR 1.11.3 Release Notes = 

The 1.11.3 release is a bug fix release.

== Build (Unix) ==
 * Support custom buildflags in GRASS plugin (#5850)
 * GRASS 7.0.0 support (#5852)
 * fix detection of ECW SDK 5.1 on Mac (#5867, #5390)
 * Add support for Poppler >= 0.31.0 (#6118)

== Port ==
 * /vsitar/: remove useless validation test that prevents from opening valid .tar files (#5864)

== Algorithms ==
 * gdal_grid: fix crash in optimized mode with GCC 4.4 on 64bit (#5987)
 * Fix failure in GDALSuggestedWarpOut2() when computing output image size in case the top-left and bottom-right corners transform to the same point (#5980)
 * Rasterize: add check to avoid burning negative x values (#5641)

== GDAL core ==
 * GMLJP2 reader: add compatibility with OGC CRS URL as found in GMLJP2v2 (#5940)
 * GMLJP2: on reading, don't do axis inversation if there's an explicit axisName requesting easting, northing order (#5960)
 * GMLJP2: add missing rangeParameters element to validate against GMLJP2 schema (#5707)

== Utilities ==
 * gdalbuildvrt: fix potential crash when using -b switch (#6095)
 * gdalserver: fix compilation with recent GNU libc (such as in Arch Linux) (#6073)
 * gdalwarp: emit error message if file specified with -cutline cannot be opened (#5921)
 * gdalwarp: fix 1.11.2 regression when invoking several times gdalwarp with several input files on the same target file, and when the input files have a nodata setting: only the last input file was warped onto the target dataset (#5909)
 * ogr2ogr: fix crash with -clipdst when a reprojection fails before (#5973)
 * ogr_layer_algebra.py: for Update, Clip and Erase, only creates attribute of input layer by default (#5976)

== GDAL drivers ==

ECRG driver:
 * change subdataset definition to make sure that they only consist of frames of same scale (#6043)

ERS driver:
 * fix SetProjection() (#5840)

GeoRaster driver:
 * New SRID search (#5911)
 * Fix user-defined SRID issue (#5881)

GTiff driver:
 * fix GTiffDataset::IsBlockAvailable() wrong behaviour when compiling against internal libtiff, when a BigTIFF file has a TileByteCounts with LONG/4-byte counts and not LONG8 (#6001)
 * avoid generated corrupted  right-most and bottom-most tiles for 12-bit JPEG-compressed (#5971)
 * on reading attach PROJ4 extension node for Mercator_Auxiliary_Sphere / WebMercator (#5924)

HDF4 driver:
 * fix AnyTypeToDouble() to use proper type (int instead of long) to work with DFNT_INT32/DFNT_UINT32 on 64-bit Linux (#5965)

HF2 driver:
 * fix reading side of the driver to work on architectures where char is unsigned, like PPC or ARM (#6082)

NetCDF driver:
 * fix crash on opening a NOAA dataset (#5962)
 * fix computation of inverse flattening (#5858)

Northwoord driver:
 * Fix computation of intermediate color table values on non-Intel platforms (#6091)
 * NWT_GRD: don't advertize scale/offset as they are transparently applied in IReadBlock() (#5839).

RPFTOC driver:
 * add tweak for weird relative directory names in the A.TOC file (#5979)

TIL driver:
 * fix half pixel shift in geo registration (#5961)

VRT driver:
 * add more checks to CheckCompatibleForDatasetIO() to avoid issues with overview bands (#5954)
 * fix rounding of output window size on VRTSimpleSource (#5874)

== OGRSpatialReference ==
 * morphFromESRI(): map Mercator_Auxiliary_Sphere to EPSG:3857 (#5924)

== OGR drivers ==

EDIGEO driver:
 * backport conversion of atof() to CPLAtof() to avoid truncation of floating point numbers including coordinates on a non C locale (#6070)

GML driver:
 * accept choice between gml:polygonProperty and gml:multiPolygonProperty (#5977)
 * fix GML_ATTRIBUTES_TO_OGR_FIELDS=YES to work correctly with xlink:href too (#5970); and also backport a trunk fix for writing the correctly path to attributes in nested elements
 * VFR: add support for OriginalniHraniceOmpv (#5853)

GeoJSON driver:
 * avoid truncation of real numbers on reading (#5882)
 * TopoJSON: fix segfault when reading https://raw.githubusercontent.com/mbostock/topojson/master/test/topojson/properties-id-computed.json (#5919)
 * GeoJSON writer: make string comparison for authority name case insensitive so as to recognize lowercase 'epsg' (#4995)

GPKG driver:
 * on reading, recognize GEOMCOLLECTION to be compatible of GDAL 2.0. Still writing GEOMETRYCOLLECTION in 1.11 branch (#5937)
 * escape column names on reading/insert/update (#5879)

MSSQLSpatial driver:
 * Add support for creating features with default values (#5957)

OpenFileGDB driver:
 * fix 1.11.2 regression with FileGDB v9 where the presence of a non spatial table can cause other layers not to be listed (#5875)

PG/PGDump drivers:
 * fix truncation of fields to work with multi-byte UTF-8 characters (#5854)

Shape driver:
 * SHPReadOGRFeature() should free the passed psShape when not NULL and if the DBF record is deleted, so as to avoid later reads to faild (linked to #5974)

SQLite driver:
 * Fix issue with consecutive identical characters in layer name with ExecuteSQL() (#6107)

== SWIG Language Bindings ==

Perl bindings:
 * Fix creation of Field Definition object (#6099)

Python bindings:
 * Fix build with SWIG 3.0.6 (#6045)

= GDAL/OGR 1.11.2 Release Notes = 

The 1.11.2 release is a bug fix release.

== Build ==

  * configure: fix detection of OCI by changing linking order to please modern GCC (#5550)
  * configure: fix test to accept MariaDB 10.X as valid MySQL (#5722)
  * More compiler warnings addressed (#5414)
  * Windows build of PDF: fix compilation issue with Visual Studio 2012 (#5744)
  * Windows build: Add support for MrSID 9.1 SDK (#5814)
  * Windows build: when building netCDF, HDF4, HDF5 as plugins, call registration of 'sub-drivers' GMT, HDF4Image and HDF5Image (#5802)

== Port ==

 * CSV finder: Stop probing for csv/horiz_cs.csv. (#5698)
 * /vsicurl/: avoid reading after end-of-file and fix failure when reading more than 16MB in a single time (#5786)
 * CPLHexToBinary(): faster implementation (#5812)


swig/csharp/apps/ogrinfo.cs
/******************************************************************************
 * $Id: ogrinfo.cs 19469 2010-04-20 20:22:11Z tamas $
 *
 * Name:     ogrinfo.cs
 * Project:  GDAL CSharp Interface
 * Purpose:  A sample app to dump information from a spatial data source.
 * Author:   Tamas Szekeres, szekerest@gmail.com
 *
 ******************************************************************************
 * Copyright (c) 2007, Tamas Szekeres
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 * DEALINGS IN THE SOFTWARE.
 *****************************************************************************/

using System;

using OSGeo.OGR;
using OSGeo.OSR;


/**

 * <p>Title: GDAL C# ogrinfo example.</p>
 * <p>Description: A sample app to dump information from a spatial data source.</p>
 * @author Tamas Szekeres (szekerest@gmail.com)
 * @version 1.0
 */



/// <summary>
/// A C# based sample to dump information from a data source.
/// </summary> 

class OGRInfo {
	
	public static void usage() 

	{ 
		Console.WriteLine("usage: ogrinfo {data source name}");
		System.Environment.Exit(-1);
	}
 
	public static void Main(string[] args) {

		if (args.Length != 1) usage();

        // Using early initialization of System.Console
        Console.WriteLine("");

		/* -------------------------------------------------------------------- */
		/*      Register format(s).                                             */
		/* -------------------------------------------------------------------- */
		Ogr.RegisterAll();

		/* -------------------------------------------------------------------- */
		/*      Open data source.                                               */
		/* -------------------------------------------------------------------- */
		DataSource ds = Ogr.Open( args[0], 0 );
		
		if (ds == null) {
			Console.WriteLine("Can't open " + args[0]);
			System.Environment.Exit(-1);
		}

		/* -------------------------------------------------------------------- */
		/*      Get driver                                                      */
		/* -------------------------------------------------------------------- */	
		Driver drv = ds.GetDriver();

		if (drv == null) 
		{
			Console.WriteLine("Can't get driver.");
			System.Environment.Exit(-1);
		}
        // TODO: drv.name is still unsafe with lazy initialization (Bug 1339)
        Console.WriteLine("Using driver " + drv.name);

		/* -------------------------------------------------------------------- */
		/*      Iterating through the layers                                    */
		/* -------------------------------------------------------------------- */	

		for( int iLayer = 0; iLayer < ds.GetLayerCount(); iLayer++ )
		{
			Layer layer = ds.GetLayerByIndex(iLayer);

			if( layer == null )
			{
				Console.WriteLine( "FAILURE: Couldn't fetch advertised layer " + iLayer );
				System.Environment.Exit(-1);
			}
			ReportLayer(layer);
		}
	}

	public static void ReportLayer(Layer layer)
	{
		FeatureDefn def = layer.GetLayerDefn();
		Console.WriteLine( "Layer name: " + def.GetName() );
		Console.WriteLine( "Feature Count: " + layer.GetFeatureCount(1) );
		Envelope ext = new Envelope();
		layer.GetExtent(ext, 1);
		Console.WriteLine( "Extent: " + ext.MinX + "," + ext.MaxX + "," +
			ext.MinY + "," + ext.MaxY);
		
		/* -------------------------------------------------------------------- */
		/*      Reading the spatial reference                                   */
		/* -------------------------------------------------------------------- */
        OSGeo.OSR.SpatialReference sr = layer.GetSpatialRef();
		string srs_wkt;
		if ( sr != null ) 
		{
			sr.ExportToPrettyWkt( out srs_wkt, 1 );
		}
		else
			srs_wkt = "(unknown)";


        Console.WriteLine( "Layer SRS WKT: " + srs_wkt );

		/* -------------------------------------------------------------------- */
		/*      Reading the fields                                              */
		/* -------------------------------------------------------------------- */
		Console.WriteLine("Field definition:");
		for( int iAttr = 0; iAttr < def.GetFieldCount(); iAttr++ )
		{
			FieldDefn fdef = def.GetFieldDefn( iAttr );
            
			Console.WriteLine( fdef.GetNameRef() + ": " + 
				fdef.GetFieldTypeName( fdef.GetFieldType() ) + " (" +
				fdef.GetWidth() + "." +
				fdef.GetPrecision() + ")");
		}

		/* -------------------------------------------------------------------- */
		/*      Reading the shapes                                              */
		/* -------------------------------------------------------------------- */
		Console.WriteLine( "" );
		Feature feat;
		while( (feat = layer.GetNextFeature()) != null )
		{
			ReportFeature(feat, def);
			feat.Dispose();
		}
	}

	public static void ReportFeature(Feature feat, FeatureDefn def)
	{
		Console.WriteLine( "Feature(" + def.GetName() + "): " + feat.GetFID() );
		for( int iField = 0; iField < feat.GetFieldCount(); iField++ )
		{
			FieldDefn fdef = def.GetFieldDefn( iField );
            
			Console.Write( fdef.GetNameRef() + " (" +
				fdef.GetFieldTypeName(fdef.GetFieldType()) + ") = ");

			if( feat.IsFieldSet( iField ) )
            {
                if (fdef.GetFieldType() == FieldType.OFTStringList)
                {
                    string[] sList = feat.GetFieldAsStringList(iField);
                    foreach (string s in sList)
                    {
                        Console.Write("\"" + s + "\" ");
                    }
                    Console.WriteLine();
                }
                else if (fdef.GetFieldType() == FieldType.OFTIntegerList)
                {
                    int count;
                    int[] iList = feat.GetFieldAsIntegerList(iField, out count);
                    for (int i = 0; i < count; i++)
                    {
                        Console.Write(iList[i] + " ");
                    }
                    Console.WriteLine();
                }
                else if (fdef.GetFieldType() == FieldType.OFTRealList)
                {
                    int count;
                    double[] iList = feat.GetFieldAsDoubleList(iField, out count);
                    for (int i = 0; i < count; i++)
                    {
                        Console.Write(iList[i].ToString() + " ");
                    }
                    Console.WriteLine();
                }
                else
                    Console.WriteLine(feat.GetFieldAsString(iField));
            }
			else
				Console.WriteLine( "(null)" );
            
		}

		if( feat.GetStyleString() != null )
			Console.WriteLine( "  Style = " + feat.GetStyleString() );
    
		Geometry geom = feat.GetGeometryRef();
		if( geom != null ) 
		{
			Console.WriteLine( "  " + geom.GetGeometryName() + 
				"(" + geom.GetGeometryType() + ")" );
			Geometry sub_geom;
			for (int i = 0; i < geom.GetGeometryCount(); i++)
			{
				sub_geom = geom.GetGeometryRef(i);
				if ( sub_geom != null )
				{
					Console.WriteLine( "  subgeom" + i + ": " + sub_geom.GetGeometryName() + 
						"(" + sub_geom.GetGeometryType() + ")" );
				}
			}
            Envelope env = new Envelope();
            geom.GetEnvelope(env);
            Console.WriteLine("   ENVELOPE: " + env.MinX + "," + env.MaxX + "," +
                env.MinY + "," + env.MaxY);

            string geom_wkt;
            geom.ExportToWkt(out geom_wkt);
            Console.WriteLine("  " + geom_wkt);
		}

		Console.WriteLine( "" );
	}
}